src/Map.php

```diff
@@ -188,6 +205,62 @@ public function diff(Map $map): Map
         });
     }
 
+    /**
+     * Returns a hashed value suitable for using as an index in an array
+     * (string or integer), trying to match the high-level behavior of the
+     * get_hash function in the extension's ds_hashtable.c. Where the extension
+     * uses the ZSTR_HASH macro to hash a string, we just use the string
+     * itself as the hash value, trusting PHP's array to handle it
+     * appropriately. Floats (doubles) are converted to an integer value by
+     * interpreting their bits as those of a long long, rewriting -0.0 to 0.0,
+     * as the extension does (but we don't xor the lower and upper 32 bits).
+     *
+     * See https://github.com/php-ds/ext-ds/blob/master/src/ds/ds_htable.c
+     *
+     * @param mixed $key Any value
+     *
+     * @return mixed The string or integer hash value.
+     *
+     * @psalm-return string|int
```

Comment:
Minor but you might be able to teach psalm about when int vs string is returned

Reply:
I don't think it matters, since we're just trying to get an `array-key`.